plugins/eval.py

```diff
@@ -56,10 +59,47 @@ def code_print(*args, sep=" ", end="\n", file=fp, flush=False):
         mk_code_print(fp)(repr(exc))
         del tb
 
+    def chunk(l, n): 
+        chunk_pos = range(0, len(l), n)
+        return [l[i:i+n] for i in chunk_pos]
+
+    # greedily concatenate strings into groups of at most a certain length
+    def chunk_concat(l, n): 
+        chunks_concat = [""] if len(l) else [] 
+        for text in l:
+            new_len = len(chunks_concat[-1] + text)
+            if new_len > n: 
+                chunks_concat.append("") 
+            chunks_concat[-1] = chunks_concat[-1] + text 
+        return chunks_concat      
+
     def format_block(fp):
         text = fp.getvalue()
         return util.discord.format("{!b:py}", text) if len(text) else "\u2705"
+    
+    def short_heuristic(fp): 
+        initial_pos = fp.tell() 
+        fp.seek(0, os.SEEK_END) 
+        fp_len = fp.tell() 
+        fp.seek(initial_pos) 
+        return fp_len <= 2000 and len(format_block(fp)) <= 2000
```

Comment:
Magic value `2000`, this will likely change in the future

Reply:
In context of discord, this is probably fine.